debian/postinst

```diff
@@ -19,32 +23,36 @@ export_gpg_key_from_shared_keyring() {
     fi
 }
 
+unconfigure_esm() {
+    rm -f $ESM_APT_GPG_KEY $ESM_APT_GPG_KEY_OLD $ESM_APT_SOURCE_FILE
+    rm -f $ESM_APT_PREF_FILE_OLD $ESM_APT_PREF_FILE
+}
+
 configure_esm() {
     rm -f $ESM_APT_GPG_KEY_OLD  # Remove retired key from key list
     if [ ! -f "$ESM_APT_GPG_KEY" ]; then
         export_gpg_key_from_shared_keyring $ESM_KEY_ID_TRUSTY
     fi
 
+    if [ -e "$ESM_APT_SOURCE_FILE_OLD" ]; then
+	mv $ESM_APT_SOURCE_FILE_OLD $ESM_APT_SOURCE_FILE
```

Comment:
I think you need to rename the preferences file too.

Reply:
We now create the apt preferences file only if a new system is installing the ESM_APT_SOURCE_FILE because only on this condition is the system both unattached and unenabled for esm.   For trusty old client with enabled ESM, we would have renamed the old apt-sources list file which contains legacy credentials already, so we do not need the preferences file pinning that source disabled to 'Pin-priority: never'.